### ISSUE TITLE
Improve prediction accuracy

### ISSUE THREAD

**Issue by pontusj101** [pontusj101/rddl_training_data_producer#21]
The prediction accuracy should be improved. Here is an example of the current state. 

This is what the model (the best-performing is a GAT) sees.
https://github.com/pontusj101/rddl_training_data_producer/assets/9636072/e26da30b-f7ca-4d85-851d-7b64e13a1a2b

This is what it predicts:
https://github.com/pontusj101/rddl_training_data_producer/assets/9636072/4c594dd8-76ca-45bf-a3ab-b5d2227622b8

And this is what actually happened:
https://github.com/pontusj101/rddl_training_data_producer/assets/9636072/f80a08e7-5eb9-433d-b18e-8df7be2b2459

Improvements can be achieved in different ways:

- Try out new architectures, such as temporal GNNs (#17).
- Modify aspects of the training data, such as the attacker policy (probably very important), the total amount of data, the amount of uncompromised snapshot sequences, etc. Many of these can be explored through hyperparameter optimization, e.g. in GCP Vertex AI.
- Train for longer, or with other training hyperparameters, such as number of layers and neurons, batch size, learning rate, etc. These can also be explored through hyperparameter optimization. I have seen in experiments that more neurons are better, so we should really try to push that limit. I think that will require GPU acceleration (#18), but we should clearly have that in any event.
